cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -33,55 +29,56 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
       message(WARNING "Unable to determine version of ${PYTHON_EXECUTABLE}: ${PYTHON_PREFER_VERSION_ERR}")
     endif()
     string(STRIP "${PYTHON_PREFER_VERSION}" PYTHON_PREFER_VERSION)
+    set(Python${PYTHON_PREFER_VERSION}_EXECUTABLE "${PYTHON_EXECUTABLE}")
   endif()
 
-  message(STATUS "Preferring Python version ${PYTHON_PREFER_VERSION}")
+  find_package(Python3 COMPONENTS Interpreter Development QUIET)
+  if(Python3_Development_FOUND)
+    # Re-run, now with NumPy, but not required:
```

Comment:
Note that the `find_package` above was not `REQUIRED` either - so looking for (optional) `NumPy` already in the first round seems to be the way to go?